pkg/validation/variables/variables_component.go

```diff
@@ -189,6 +195,81 @@ func validateAndReplaceForOpenShiftComponent(variables map[string]string, opensh
 	return newInvalidKeysError(invalidKeys)
 }
 
+// validateAndReplaceForImageComponent validates the image component data for global variable references and replaces them with the variable value
+func validateAndReplaceForImageComponent(variables map[string]string, image *v1alpha2.ImageComponent) error {
+	var err error
+
+	invalidKeys := make(map[string]bool)
+
+	if image != nil {
+		// Validate image's image name
+		if image.ImageName, err = validateAndReplaceDataWithVariable(image.ImageName, variables); err != nil {
+			checkForInvalidError(invalidKeys, err)
+		}
+
+		switch {
```

Comment:
why simple if statement would be worse here?)
I assume it's written in that way for future when another image types appear.
I don't have a strong opinion but personally, I would go with if until there is no case for switch.

Reply:
updated